app/scripts/modules/core/src/pipeline/config/triggers/triggers.html

```diff
@@ -15,6 +15,12 @@
             <help-field key="pipeline.config.parallel.cancel.queue"></help-field>
           </label>
         </div>
+        <div class="checkbox">
+          <label>
+            <input type="checkbox" ng-model="pipeline.pinAllParameters" />
+            <strong>Pin all parameters when viewing an execution</strong>
```

Comment:
there's a prior comment on this line:
https://github.com/armory-io/deck/pull/4#issuecomment-478275517
<kbd>![](https://cl.ly/4ae2c0c72ff3/download/Screen%20Shot%202019-04-01%20at%2017.39.16.png)</kbd>

> Would it be fair to say Permalink defaults to expanded, but the user can still click the link to collapse? That's what the screenshot seems to show and is also what I think I would prefer.

YEP!

> I think I'd prefer to see "pin all parameters when viewing an execution" replaced with a "pin all"/"unpin all" toggle in the Parameters section. @gcomstock if you're in the office next week, can you provide some guidance on the direction you'd like to go for check/uncheck all in the forms mockups?

hm.. I can get behind this. A mockup would be great! I'm not sure I have a good idea myself on it.

Reply:
Hey @kevinawoo id be very happy to get you a mockup. Im crunching on something else at the moment but will make a note to do this later today.

Reply:
oh would you also be able to get me a mockup with the `pin/unpin all artifacts` too? you'll may need to set `window.spinnakerSettings.feature.artifactsRewrite = true;` in `settings-local.js`.

But there's some new UI elemenets with it:
<kbd>![](https://cl.ly/2585ea7bfd48/download/Screen%20Shot%202019-04-03%20at%2011.22.21.png)</kbd>
<kbd>![](https://cl.ly/4aea71f4d16c/download/Screen%20Shot%202019-04-03%20at%2011.22.48.png)</kbd>
<kbd>![](https://cl.ly/f89b1384eb43/download/Screen%20Shot%202019-04-03%20at%2011.23.52.png)</kbd>

Reply:
I think we should just leave the old style of artifacts alone for now, since the inline style will be enabled by default later.

Reply:
another option is just to merge this and come back to artifacts later, after there's a need. We don't currently have a need to pin artifacts yet, parameters are the ones that are exploding at the moment